### PR TITLE
Fixing cache when ID changes

### PR DESCRIPTION
**Pull request by caiosba**
## Description

The URL ID, used as a cache key, is calculated in runtime, at different times, when writing and when reading. This means that the cache is written to a key but read from another one. This is a pretty edge case, but this PR turns it safer and more consistent.

Reference: CV2-3749.

## How has this been tested?

TDD. I added a test that first reproduced the issue. The test passed after the fix was implemented.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [x] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [x] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: test/controllers/medias_controller_test.rb [hunk 1]
```diff
@@ -686,6 +686,22 @@ def setup
     assert_equal url, JSON.parse(@response.body)['data']['title']
     assert_equal Lapis::ErrorCodes::const_get('DUPLICATED'), JSON.parse(@response.body)['data']['error']['code']
   end
+
+  test "should refresh cache even if ID changes" do
+    WebMock.enable!
+    WebMock.disable_net_connect!(allow: [/minio/])
+    Media.stubs(:get_id).returns('foo', 'bar', 'foo', 'bar')
```
Comment (vasconsaurus) [inline test/controllers/medias_controller_test.rb:694]: @caiosba if/when you have the time, would you mind explaining how line 693 works again?

Comment (caiosba) [reply test/controllers/medias_controller_test.rb:694]: Sure! It basically means that each time the stubbed method (in this case, `get_id`) is called during that test, the returned value will be different. So, first time it will be `foo`, second time it will be `bar`.

Comment (vasconsaurus) [reply test/controllers/medias_controller_test.rb:694]: why `.returns('foo', 'bar', 'foo', 'bar')` and not `.returns('foo', 'bar')` ?

Comment (caiosba) [reply test/controllers/medias_controller_test.rb:694]: Because it runs four times during that test, 2 in the controller and 2 in the model. I can show you when we pair and explain.
